embedded-graphics/src/primitives/line.rs

```diff
@@ -65,76 +60,318 @@ impl Dimensions for Line {
 impl Line {
     /// Create a new line
     pub const fn new(start: Point, end: Point) -> Self {
-        Line { start, end }
+        Self { start, end }
+    }
+}
+
+/// Which side of the center line to draw on
+///
+/// Imagine standing on `start`, looking ahead to where `end` is. `Left` is to your left, `Right` to
+/// your right.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub enum Side {
+    /// DOCS LOL
+    Left,
+    /// DOCS LOL
+    Right,
+}
+
+impl Side {
+    fn swap(&mut self) {
+        match self {
+            Self::Left => *self = Self::Right,
+            Self::Right => *self = Self::Left,
+        }
+    }
+}
+
+/// Current state of each parallel line drawn
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+struct ParallelLineState {
+    start: Point,
+    error: i32,
+    dx_accum: u32,
+}
+
+impl ParallelLineState {
+    fn new(start: Point, initial_error: i32, dx_accum: u32) -> Self {
+        Self {
+            start,
+            error: initial_error,
+            dx_accum,
+        }
+    }
+}
+
+/// Current side state
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+struct SideState {
+    /// Parallel line start point
+    start: Point,
```

Comment:
See comment above.

Reply:
Changed to `parallel_start`, because it's used as the starting point of a `ParallelLineState`. Open to better suggestions :)